smelter-api/src/output/whip_into.rs

```diff
@@ -19,120 +17,33 @@ impl TryFrom<WhipOutput> for core::RegisterOutputOptions {
-                        WhipVideoEncoderOptions::FfmpegVp9 {
-                            pixel_format,
-                            ffmpeg_options,
-                        } => {
-                            vec![core::VideoEncoderOptions::FfmpegVp9(
-                                core::FfmpegVp9EncoderOptions {
-                                    resolution: options.resolution.clone().into(),
-                                    pixel_format: pixel_format
-                                        .unwrap_or(PixelFormat::Yuv420p)
-                                        .into(),
-                                    raw_options: ffmpeg_options
-                                        .unwrap_or_default()
-                                        .into_iter()
-                                        .collect(),
-                                },
-                            )]
-                        }
-                        WhipVideoEncoderOptions::Any => {
-                            vec![
-                                core::VideoEncoderOptions::FfmpegVp9(
-                                    core::FfmpegVp9EncoderOptions {
-                                        resolution: options.resolution.clone().into(),
-                                        pixel_format: core::OutputPixelFormat::YUV420P,
-                                        raw_options: Vec::new(),
-                                    },
-                                ),
-                                core::VideoEncoderOptions::FfmpegVp8(
-                                    core::FfmpegVp8EncoderOptions {
-                                        resolution: options.resolution.clone().into(),
-                                        raw_options: Vec::new(),
-                                    },
-                                ),
-                                core::VideoEncoderOptions::FfmpegH264(
-                                    core::FfmpegH264EncoderOptions {
-                                        preset: H264EncoderPreset::Fast.into(),
-                                        resolution: options.resolution.clone().into(),
-                                        pixel_format: core::OutputPixelFormat::YUV420P,
-                                        raw_options: Vec::new(),
-                                    },
-                                ),
-                            ]
-                        }
-                    };
-                    Ok(encoder_options)
-                })
-                .try_collect()?;
+                let encoder_preferences = match options.encoder_preferences.as_deref() {
+                    Some([]) | None => {
+                        vec![core::WhipVideoEncoderOptions::Any(resolution.into())]
+                    }
+                    Some(prefs) => prefs
+                        .iter()
+                        .cloned()
+                        .map(|p| p.into_pipeline_options(resolution.clone()))
```

Comment:
```suggestion
                        .map(|p| p.into_pipeline_options(resolution))
```
resolution is or can be Copy